packages/vats/src/localchain.js

```diff
@@ -67,20 +124,24 @@ const prepareLocalChainAccount = zone =>
        * @param {Payment<'nat'>} payment
        * @param {Pattern} [optAmountShape] throws if the Amount of the Payment
        *   does not match the provided Pattern
-       * @returns {Promise<Amount>}
+       * @returns {PromiseVow<Amount<'nat'>>}
        */
       async deposit(payment, optAmountShape) {
         const { bank } = this.state;
 
-        const allegedBrand = await E(payment).getAllegedBrand();
-        const purse = E(bank).getPurse(allegedBrand);
-        return E(purse).deposit(payment, optAmountShape);
+        const depositPaymentHandler = makeDepositPaymentHandler();
```

Comment:
I thought that having a single `depositPaymentHandler` per `LocalChain` instance actually made sense. For that matter it could be a facet of a kit. `bank` would come from state, and `payment` and `optAmountShape` from bound values.

Reply:
> I thought that having a single depositPaymentHandler per LocalChain instance actually made sense
The revision where it was in state? 

> For that matter it could be a facet of a kit. 
We considered a kit but thought this approach would result in fewer refactors. Can give a `kit` a go.

Reply:
Yeah a singleton in state or a facet, both work IMO. No need to allocate a new deposit handler per invocation.